### PR TITLE
Add team-type feature flag for Shared Library

### PR DESCRIPTION
**Pull request by knolleary**
## Description

Part of #2531 

This adds a flag to the TeamType to control whether the Shared Library feature is available.

It updates the settings end point used by instances to get their configuration to include a features object. This is the server side of https://github.com/flowforge/flowforge-nr-launcher/pull/156

Adds a new banner to display when accessing a feature that is not available at the team level. It includes the invite to upgrade your team - but we don't have a page for that yet.

image

### CODE REVIEW

File: frontend/src/components/banners/FeatureUnavailableToTeam.vue [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<template>
+    <div
+        class="ff-page-banner my-4"
+        data-el="page-banner-feature-unavailable-to-team"
+    >
+        <SparklesIcon class="ff-icon mr-2" style="stroke-width: 1px;" />
+        <div>
+            This feature is not available for your current Team. Please <a class="ff-link" href="#" target="_blank" rel="noopener noreferrer">upgrade</a> your Team in order to use it.
```
[the remark this answers is not in the document]
Comment (knolleary) [reply frontend/src/components/banners/FeatureUnavailableToTeam.vue:8]: See this comment in the description:
> Adds a new banner to display when accessing a feature that is not available at the team level. It includes the invite to upgrade your team - but we don't have a page for that yet.

That will get filled in once we have the upgrade path in the UI

[the remark this answers is not in the document]
Comment (Steve-Mcl) [reply frontend/src/components/banners/FeatureUnavailableToTeam.vue:8]: I get that, my point was more why are we setting an anchor when there is nothing to point to?

If we ship this, then the link will do nothing & be odd to the user.  

Is it not better to have no link until the destination is ready?

Or am I missing something?

[the remark this answers is not in the document]
Comment (Pezmc) [reply frontend/src/components/banners/FeatureUnavailableToTeam.vue:8]: I agree with @Steve-Mcl, this should block the PR from being merged until we do have an upgrade link, or the `a` removed entirely

**Review by Pezmc — APPROVED** [review]

File: frontend/src/components/banners/FeatureUnavailableToTeam.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (knolleary) [reply frontend/src/components/banners/FeatureUnavailableToTeam.vue]: FYI, this is updated in https://github.com/flowforge/flowforge/pull/2634 which targets the branch of this PR. This PR will be the last one to merge.

**Pull request merged**